### ISSUE TITLE
Remove share widget exclusion for elife

### ISSUE THREAD

**Issue by dwhly** [hypothesis/product-backlog#882]
### Problem you are trying to address with this feature

There is no share button for the elife implementation.  

This commit:
https://github.com/hypothesis/client/commit/d11be3fb0ac168d80c65603bad4b735bd455200b

Removed this capability, but only for their authority.  Per the commit message:

> Don't show the "Share this page" button in the top-right of the sidebar
> on pages, such as eLife's, that have both an embedded Hypothesis client
> and a services config including their own `authority` string.
> 
> The "Share this page" button is a Via link, and Via doesn't work with
> pages using a third-party authority config.

### Your solution

Now that we use bouncer links in the share widget, as of this PR https://github.com/hypothesis/client/pull/748  and also now that we can whitelist sites embedding H per this PR https://github.com/hypothesis/bouncer/pull/155, we should remove this exception.

Also, note that we will presumably need to add elife to the bouncer/embed_detector.py at the same time.

**Comment by robertknight** [comment]
To clarify, this is about the "share this page" toolbar button at the top of the client which normally contains an https://hyp.is/go link. Individual annotation cards on eLife pages still have share buttons. In the context of eLife though, the link to share the page is just ... the URL of the page itself, so the "Share" link in the toolbar is a bit pointless in a sense.

The one thing the share link does do which won't happen if you share just the URL of the page itself is that it causes the sidebar to be opened automatically. This won't happen if bouncer naively just redirects to the URL of the page.

**Comment by dwhly** [comment]
>  To clarify, this is about the "share this page" toolbar button at the top of the client which normally contains an https://hyp.is/go link.

Yes.  This is the "Share all the annotations on this page" with someone else link, as opposed to the "Share this annotation" with someone else link.

>  In the context of eLife though, the link to share the page is just ... the URL of the page itself, so the "Share" link in the toolbar is a bit pointless in a sense.

There's a couple reasons to always preserve the share link functionality:
1- So that people can rely on always being able to do the same thing in the same place-- and not worry about which kind of H instance this is, or whether the page has H embedded or not.  These are distinctions they won't know or understand, nor should they have to.
2- We're imagining that once we can share groups in bouncer links that the share link will be scoped to what the group scope control is.  So that you're sharing "the annotations you're looking at right now" over this page.  Of course, there should be text around the link you copy that indicates how exactly it's scoped, so that it's clear to end users.
3- One could extend that argument to include "any active search within the sidebar", for instance-- if a particular user is being searched in the sidebar, that the share functionality would scope the bouncer link to the active search (including the active group).  Again, any of this would need to be explained with text to users that are grabbing the share link, so that they understand.
4- The use of the bouncer link also has the effect of opening the sidebar.  This is probably a critical aspect of the sharing UX as well.

**Comment by klemay** [comment]
> 4- The use of the bouncer link also has the effect of opening the sidebar. This is probably a critical aspect of the sharing UX as well.

Strongly agree here. As Heather mentioned in an email, there is a crucial difference between sharing a link to the page and sharing a link to the page _with the annotation sidebar enabled and open_. For partners like eLife who have H disabled until the user clicks a button, this is going to be especially important.

**Comment by dwhly** [comment]
> For partners like eLife who have H disabled until the user clicks a button

Good point, hadn't remembered that part.

**Comment by klemay** [comment]
I talked with @ajpeddakotla today since there was confusion about what actually needs to be done for this ticket and why we're doing it.

## Why we're doing it:
- It's useful to have the "share this link" toolbar button _in addition to the individual share links on annotation cards_. All of our non eLife users have it.
- We only took this button away because previously, the `https://hyp.is/go` link generated by the button caused eLife users to get redirected to Via, which was a poor user experience
- Since https://github.com/hypothesis/bouncer/pull/155 was implemented, we can ensure bouncer doesn't redirect eLife users to Via

In short, we're restoring functionality that we only took away because of an undesirable UX that we now have a workaround for.

## What needs to be done:


### 1. Bring the "share this page" button back to the toolbar for eLife:

![screenshot-web hypothes is-2019 01 31-14-02-38](https://user-images.githubusercontent.com/12087705/52078392-17088300-2561-11e9-96d0-9d007ab58b03.png)

This will require @robertknight or another dev.

### 2. Add eLife to the list of URLs for which bouncer does not redirect users to Via. 

I can do this once Part 1 is done if you notify me!

**Comment by robertknight** [comment]
@klemay - Can you confirm which URL patterns eLife include Hypothesis on? Is it all of https://elifesciences.org or only a subset of URLs under certain paths?

Something we'll need to figure out here is how to enable the share functionality for eLife but avoid showing the share button in other third-party contexts where it won't work, such as when the document is embedded by the LMS app. A really dumb approach might be to require the publisher to explicitly enable the page sharing functionality via a config option.

**Comment by robertknight** [comment]
If the client can determine that it was embedded by the publisher rather than by a browser extension or proxy service, it could enable the share dialog and provide some hint to bouncer in the URL. Bouncer could then choose to respect or ignore that hint.

**Comment by klemay** [comment]
@robertknight I believe it's embedded on all articles but not landing pages, so anything with `https://elifesciences.org/articles/`

**Comment by robertknight** [comment]
We didn't get to this in the last sprint as other work took priority. This is assigned to me still to resolve one way or another.

**Comment by dwhly** [comment]
> If the client can determine that it was embedded by the publisher rather than by a browser extension or proxy service, it could enable the share dialog and provide some hint to bouncer in the URL. Bouncer could then choose to respect or ignore that hint.

I think there are three steps here:

1. Whitelist elifesciences as having H embedded. (embed_detector.py?)
2. Remove the particular bit of code that says "if this site is elifesciences, then don't show the share box."
3. Then, perhaps later as you suggest, a fancier next step so that our code passes on when it's embedded through the bouncer link (though of course that would suffer from a time lag issue if at the time that the URL was formed H wasn't embedded and then later it was-- for instance if the URL was encoded as a link in some HTML somewhere).  

But presumably we're not going to determine *whether or not* to show the share box in this last case, only how to format a fancier bouncer link.  We should always show the share box from now on after we do step 2 above.
